fluentd/run.sh

```diff
@@ -101,6 +104,33 @@ else
     rm -f $CFG_DIR/openshift/filter-pre-mux-client.conf
 fi
 
+if [ "${USE_MUX:-}" = "true" ] ; then
+    TOTAL_MEMORY_LIMIT=`echo $MUX_MEMORY_LIMIT |  sed -e "s/[Kk]/*1024/g;s/[Mm]/*1024*1024/g;s/[Gg]/*1024*1024*1024/g;s/i//g" | bc`
+else
+    TOTAL_MEMORY_LIMIT=`echo $FLUENTD_MEMORY_LIMIT |  sed -e "s/[Kk]/*1024/g;s/[Mm]/*1024*1024/g;s/[Gg]/*1024*1024*1024/g;s/i//g" | bc`
+fi
+BUFFER_SIZE_LIMIT=`echo $BUFFER_SIZE_LIMIT |  sed -e "s/[Kk]/*1024/g;s/[Mm]/*1024*1024/g;s/[Gg]/*1024*1024*1024/g;s/i//g" | bc`
+if [ $BUFFER_SIZE_LIMIT -eq 0 ]; then
+    BUFFER_SIZE_LIMIT=1048576
+fi
+
+DIV=1
+if [ "$ES_HOST" != "$OPS_HOST" ] || [ "$ES_PORT" != "$OPS_PORT" ] ; then
+    # using ops cluster
+    DIV=`expr $DIV \* 2`
+fi
+if [ "${USE_MUX_CLIENT:-}" = "true" ] ; then
+    DIV=`expr $DIV \* 2`
```

Comment:
The more we discuss this, the more I wonder if we should simplify this a bit:
 * always use mux
 * fluentd clients have one queue to mux
 * fluentd output queue is `file` based
 * fluentd clients have a large chunk size (16MB), queue size of 100 (default), both settable
 * fluentd memory size calculated to be enough to run ruby
 * mux has one queue per ES instance
 * mux output queue(s) are `file` based
 * mux also have large chunk size (32 MB, a bit bigger for ES payload efficiency), settable
 * mux queue size defaults to 100, settable
 * mex memory size calculated to be enough to run ruby